raft/logger.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type Logger interface {
+	// EnableTimestamps enables the logger to print timestamp
+	// prefix. It is useful to disable it when using journald.
+	EnableTimestamps()
+	// EnableDebug enables the logger to print out debugging
+	// information. It can be very verbose.
+	EnableDebug()
```

Comment:
Again, this seems like the wrong level for this. In `glog`, verbose output is controlled by command-line flags; we can't set it in the application. I understand you need some way of turning on debug output for the standard log package, but I don't think it should be in the interface that all loggers are required to implement.